evaluate/task/task-write-test.go

```diff
@@ -43,7 +43,7 @@ func (t *TaskWriteTests) Identifier() evaltask.Identifier {
 }
 
 // TaskWriteTests generates test files for the given implementation file in a repository.
-func (t *TaskWriteTests) Run(repository evaltask.Repository) (repositoryAssessment metrics.Assessments, problems []error, err error) {
+func (t *TaskWriteTests) Run(repository evaltask.Repository) (repositoryAssessment map[evaltask.Identifier]metrics.Assessments, problems []error, err error) {
```

Comment:
Have you thought about passing the assessment store to `TaskWriteTests.Run` as parameter, and letting the the assessments be stored directly in there instead of returning a map of assessments?

I introduced the assessment store to get rid of the `map[x]map[y]map[z]....`. It feels a bit like we are starting that again: starting with a single level map, but then we need more and it becomes again a `map[x]map[y]map[z]...` thing.

This is nothing you need to change now if this already works, but something to keep in mind please.

Reply:
Think we tried that but Rui got a cyclic import.